pyopencl/array.py

```diff
@@ -81,88 +75,19 @@ def _get_common_dtype(obj1, obj2, queue):
         raise ValueError("PyOpenCL array has no queue; call .with_queue() to "
                 "add one in order to be able to perform operations")
 
-    allow_double = has_double_support(queue.device)
-    cache_key = None
-    o1_is_array = isinstance(obj1, Array)
-    o2_is_array = isinstance(obj2, Array)
-
-    if o1_is_array and o2_is_array:
-        o1_dtype = obj1.dtype
-        o2_dtype = obj2.dtype
-        cache_key = (obj1.dtype, obj2.dtype, allow_double)
-    else:
-        o1_dtype = getattr(obj1, "dtype", type(obj1))
-        o2_dtype = getattr(obj2, "dtype", type(obj2))
-
-        o1_is_integral = np.issubdtype(o1_dtype, np.integer)
-        o2_is_integral = np.issubdtype(o1_dtype, np.integer)
-
-        o1_key = obj1 if o1_is_integral and not o1_is_array else o1_dtype
-        o2_key = obj2 if o2_is_integral and not o2_is_array else o2_dtype
+    # Note: We are calling np.result_type with pyopencl arrays here.
+    # Luckily, np.result_type only looks at the dtype of input arrays as of
+    # numpy v2.1.
```

Comment:
This needs a bump in `pyproject.toml` to `numpy>=2.1` if true?

Reply:
Hmm... then this would put this PR on ice until we can actually depend on numpy 2.1? numpy 2.1 requires Py3.10, and, at least for now, the compat target is at 3.8. 3.8 [hits EOL](https://devguide.python.org/versions/) this month, so maybe this would justify a compat bump to 3.10 (globally, across all packages)?

Reply:
What this comment intends to say is "np.result_type only looks at the dtype of input arrays until at least numpy v2.1". I tested this with numpy 1.23, and it seems to work fine.

Reply:
Clarified the comment a bit in 0c37b58

Reply:
Ah, I see! Then, yeah, that sounds completely fine!